🐛 Bug: Internal Server Error - reflection on nil

### Reproduction steps

When using trafik + oidc + authentik it will sometimes return a simple html page with 'Internal Server Error' on the page. Likely coming from traefik itself.

This happens periodically, every couple of minutes and stays that way for a few seconds and then resolves itself. Then happens again a few minutes later.

### Expected vs actual behavior

It would be nice to have consistent behavior and be able to access my containers as expected.

### Environment and Version Numbers

Traefik v.3.4 had the issue, realized it was a bit old and upgraded to v3.6.1 and that still had the issue
Plugin: v17 (i don' recall the timing of this upgrade and it might be that v16 was working as expected)
Authentik: 2025.10.1

I am using LetsEncrypt and all my services are being access through this combo and are all running in docker.

### Log Output

Here is an example log from the traefik container when an error happens:

2025-11-17T11:52:59-08:00 ERR plugins-storage/sources/gop-2038983719/src/github.com/sevensolutions/traefik-oidc-auth/src/main.go:108:5: panic: github.com/sevensolutions/traefik-oidc-auth/src.ServeHTTP(...) module=github.com/sevensolutions/traefik-oidc-auth plugin=plugin-traefik-oidc-auth runtime=
2025-11-17T11:52:59-08:00 ERR Recovered from panic in HTTP handler [192.168.9.229:57466 - https://foo.example.com/bar]: reflect: Method on nil interface value middlewareName=traefik-internal-recovery middlewareType=Recovery
2025-11-17T11:52:59-08:00 ERR Stack: goroutine 5898 [running]:
github.com/traefik/traefik/v3/pkg/middlewares/recovery.recoverFunc({0x78f63a0, 0xc0067465b8}, 0xc0064ab400)
	github.com/traefik/traefik/v3/pkg/middlewares/recovery/recovery.go:52 +0x231
panic({0x5dad540?, 0x7869df0?})
	runtime/panic.go:792 +0x132
github.com/traefik/yaegi/interp.runCfg.func1()
	github.com/traefik/yaegi@v0.16.1/interp/run.go:226 +0x1ae
panic({0x5dad540?, 0x7869df0?})
	runtime/panic.go:792 +0x132
reflect.Value.Method({0x620ac40?, 0xc005401ba0?, 0x5f?}, 0x0)
	reflect/value.go:1810 +0x13c
github.com/traefik/yaegi/interp.getIndexBinMethod.func1(0xc00430e9a0?)
	github.com/traefik/yaegi@v0.16.1/interp/run.go:1683 +0xa8
github.com/traefik/yaegi/interp.runCfg(0xc003405680, 0xc00430e9a0, 0xc0019265e0?, 0x6bfe820?)
	github.com/traefik/yaegi@v0.16.1/interp/run.go:234 +0x265
github.com/traefik/yaegi/interp.genFunctionWrapper.func1.1({0xc0010a32f0, 0x2, 0x4?})
	github.com/traefik/yaegi@v0.16.1/interp/run.go:1042 +0x805
github.com/traefik/yaegi/stdlib._net_http_Handler.ServeHTTP(...)
	github.com/traefik/yaegi@v0.16.1/stdlib/go1_22_net_http.go:296
github.com/traefik/traefik/v3/pkg/server/middleware.(*traceablePlugin).ServeHTTP(0x78f43d8?, {0x7252f6747cd0?, 0xc0067465b8?}, 0x0?)
	github.com/traefik/traefik/v3/pkg/server/middleware/plugins.go:54 +0x26
github.com/traefik/traefik/v3/pkg/middlewares/observability.(*middlewareTracing).ServeHTTP(0xc0010393b0, {0x7252f6747cd0, 0xc0067465b8}, 0xc0064ab7c0)
	github.com/traefik/traefik/v3/pkg/middlewares/observability/middleware.go:67 +0x4a9
github.com/traefik/traefik/v3/pkg/middlewares/accesslog.(*ConcatFieldHandler).ServeHTTP(0xc001039410, {0x7252f6747cd0, 0xc0067465b8}, 0xc0064ab7c0)
	github.com/traefik/traefik/v3/pkg/middlewares/accesslog/field_middleware.go:98 +0x247
github.com/traefik/traefik/v3/pkg/middlewares/observability.(*routerTracing).ServeHTTP(0xc0034f3580, {0x7252f6747cd0, 0xc0067465b8}, 0xc0064ab7c0)
	github.com/traefik/traefik/v3/pkg/middlewares/observability/router.go:59 +0x763
github.com/traefik/traefik/v3/pkg/middlewares/observability.(*semConvServerMetrics).ServeHTTP(0x5bd7d80?, {0x7252f6747cd0?, 0xc0067465b8?}, 0x4c83e5?)
	github.com/traefik/traefik/v3/pkg/middlewares/observability/semconv.go:50 +0xeb8
github.com/traefik/traefik/v3/pkg/middlewares/accesslog.(*FieldHandler).ServeHTTP(0xc0034f3640, {0x7252f6747cd0, 0xc0067465b8}, 0xc0064ab7c0)
	github.com/traefik/traefik/v3/pkg/middlewares/accesslog/field_middleware.go:33 +0x11e
github.com/traefik/traefik/v3/pkg/middlewares/accesslog.(*Handler).ServeHTTP(0xc0009002a0, {0x7252f6747cd0, 0xc0067465b8}, 0xc0064ab7c0, {0x78784c0, 0xc0034f3640})
	github.com/traefik/traefik/v3/pkg/middlewares/accesslog/logger.go:212 +0x13ef
github.com/traefik/traefik/v3/pkg/server/middleware.(*ObservabilityMgr).BuildEPChain.(*Handler).AliceConstructor.func4.1({0x7252f6747cd0?, 0xc0067465b8?}, 0x78f4930?)
	github.com/traefik/traefik/v3/pkg/middlewares/accesslog/logger.go:86 +0x3a
net/http.HandlerFunc.ServeHTTP(0xc0010a32c0?, {0x7252f6747cd0?, 0xc0067465b8?}, 0x0?)
	net/http/server.go:2294 +0x29
github.com/traefik/traefik/v3/pkg/middlewares/observability.(*entryPointTracing).ServeHTTP(0xc001039530?, {0x7252f6747cd0?, 0xc0067465b8?}, 0xc0064ab7c0?)
	github.com/traefik/traefik/v3/pkg/middlewares/observability/entrypoint.go:52 +0x25e
github.com/traefik/traefik/v3/pkg/server/middleware.(*ObservabilityMgr).observabilityContextHandler.WithObservabilityHandler.func1({0x7252f6747cd0, 0xc0067465b8}, 0xc0064ab540)
	github.com/traefik/traefik/v3/pkg/middlewares/observability/observability.go:28 +0x97
net/http.HandlerFunc.ServeHTTP(0xc00331dd80?, {0x7252f6747cd0?, 0xc0067465b8?}, 0x8?)
	net/http/server.go:2294 +0x29
github.com/traefik/traefik/v3/pkg/muxer/http.(*Muxer).ServeHTTP(0xc000de9f50, {0x7252f6747cd0, 0xc0067465b8}, 0xc0064ab400)
	github.com/traefik/traefik/v3/pkg/muxer/http/mux.go:53 +0x151
github.com/traefik/traefik/v3/pkg/middlewares/recovery.(*recovery).ServeHTTP(0xc004955ac0, {0x78d8038, 0xc0067465b0}, 0xc0064ab400)
	github.com/traefik/traefik/v3/pkg/middlewares/recovery/recovery.go:36 +0x147
github.com/traefik/traefik/v3/pkg/middlewares/snicheck.SNICheck.ServeHTTP({{0x7882800?, 0xc004955ac0?}, 0xc001094180?}, {0x78d8038, 0xc0067465b0}, 0xc0064ab400)
	github.com/traefik/traefik/v3/pkg/middlewares/snicheck/snicheck.go:49 +0x162
github.com/traefik/traefik/v3/pkg/middlewares.(*HTTPHandlerSwitcher).ServeHTTP(0x41c765?, {0x78d8038, 0xc0067465b0}, 0xc0064ab400)
	github.com/traefik/traefik/v3/pkg/middlewares/handler_switcher.go:23 +0x91
github.com/traefik/traefik/v3/pkg/middlewares/requestdecorator.(*RequestDecorator).ServeHTTP(0xc00013c270, {0x78d8038, 0xc0067465b0}, 0xc0064ab2c0, 0xc0064c3970)
	github.com/traefik/traefik/v3/pkg/middlewares/requestdecorator/request_decorator.go:47 +0x2ee
github.com/traefik/traefik/v3/pkg/server.newHTTPServer.WrapHandler.func5.1({0x78d8038?, 0xc0067465b0?}, 0xc0010a3170?)
	github.com/traefik/traefik/v3/pkg/middlewares/requestdecorator/request_decorator.go:89 +0x53
net/http.HandlerFunc.ServeHTTP(0xc0009005a0?, {0x78d8038?, 0xc0067465b0?}, 0x9?)
	net/http/server.go:2294 +0x29
github.com/traefik/traefik/v3/pkg/middlewares/forwardedheaders.(*XForwarded).ServeHTTP(0xc0009005a0, {0x78d8038, 0xc0067465b0}, 0xc0064ab2c0)
	github.com/traefik/traefik/v3/pkg/middlewares/forwardedheaders/forwarded_header.go:201 +0xc2
github.com/traefik/traefik/v3/pkg/server.newHTTPServer.DisableAutoDetection.func7({0x78d8038, 0xc0067465b0}, 0xc0064ab2c0)
	github.com/traefik/traefik/v3/pkg/middlewares/contenttype/content_type.go:56 +0xc5
net/http.HandlerFunc.ServeHTTP(0x0?, {0x78d8038?, 0xc0067465b0?}, 0x689005?)
	net/http/server.go:2294 +0x29
github.com/traefik/traefik/v3/pkg/server.newHTTPServer.AllowQuerySemicolons.func9({0x78d8038, 0xc0067465b0}, 0xc0064ab2c0)
	net/http/server.go:3325 +0x1cf
net/http.HandlerFunc.ServeHTTP(0xc004c9f7a0?, {0x78d8038?, 0xc0067465b0?}, 0x5?)
	net/http/server.go:2294 +0x29
github.com/traefik/traefik/v3/pkg/server.newHTTPServer.sanitizePath.func10({0x78d8038, 0xc0067465b0}, 0xc0064ab180)
	github.com/traefik/traefik/v3/pkg/server/server_entrypoint_tcp.go:820 +0x11c
net/http.HandlerFunc.ServeHTTP(0x491ebd?, {0x78d8038?, 0xc0067465b0?}, 0x40df4e?)
	net/http/server.go:2294 +0x29
github.com/traefik/traefik/v3/pkg/server.newHTTPServer.normalizePath.func11({0x78d8038?, 0xc0067465b0?}, 0xf?)
	github.com/traefik/traefik/v3/pkg/server/server_entrypoint_tcp.go:856 +0x94
net/http.HandlerFunc.ServeHTTP(0x0?, {0x78d8038?, 0xc0067465b0?}, 0x0?)
	net/http/server.go:2294 +0x29
github.com/traefik/traefik/v3/pkg/server.newHTTPServer.denyFragment.func12({0x78d8038, 0xc0067465b0}, 0xc0064ab180)
	github.com/traefik/traefik/v3/pkg/server/server_entrypoint_tcp.go:803 +0xee
net/http.HandlerFunc.ServeHTTP(0xc0030826c0?, {0x78d8038?, 0xc0067465b0?}, 0xc002eb9000?)
	net/http/server.go:2294 +0x29
net/http.serverHandler.ServeHTTP({0x6df704?}, {0x78d8038?, 0xc0067465b0?}, 0xc002eb9000?)
	net/http/server.go:3301 +0x8e
net/http.initALPNRequest.ServeHTTP({{0x78f43d8?, 0xc001373860?}, 0xc004cc2e08?, {0xc0001ff700?}}, {0x78d8038, 0xc0067465b0}, 0xc0064ab180)
	net/http/server.go:3974 +0x231
net/http.(*http2serverConn).runHandler(0x4501f2?, 0x0?, 0x0?, 0xc000ce0ff0?)
	net/http/h2_bundle.go:6529 +0xf5
created by net/http.(*http2serverConn).scheduleHandler in goroutine 2157
	net/http/h2_bundle.go:6463 +0x21d
 middlewareName=traefik-internal-recovery middlewareType=Recovery

### Doing your homework

- [x] I have searched for similar bug reports and not found a similar one.

--- a/src/session.go
+++ b/src/session.go
@@ -160,6 +160,9 @@ func validateSessionTicket(toa *TraefikOidcAuth, encryptedTicket string) (*sessi
 
 			return session, claims, session, err
 		} else {
+			if err == nil {
+				err = errors.New("no refresh_token available")
+			}
 			return nil, nil, nil, err
 		}
 	}

